File: connectors/connector-jira/src/main/scala/com/taskadapter/connector/jira/JiraTaskTypeResolver.scala
package com.taskadapter.connector.jira

object JiraTaskTypeResolver {

  def resolveIssueTypeNameForCreate(wrapper: IssueWrapper, defaultTaskTypeName: String, defaultIssueTypeForSubtasks: String): String = {
    if (wrapper.taskType.isDefined && !wrapper.taskType.get.isEmpty) {
      wrapper.taskType.get
    } else {
      if (wrapper.issueInput.getField("parent") == null) defaultTaskTypeName
      else defaultIssueTypeForSubtasks
    }
  }
}

File: connectors/connector-jira/src/main/java/com/taskadapter/connector/jira/JiraTaskTypeResolver.java
package com.taskadapter.connector.jira;

public class JiraTaskTypeResolver {
    public static String resolveIssueTypeNameForCreate(IssueWrapper wrapper,
                                                       String defaultTaskTypeName,
                                                       String defaultIssueTypeForSubtasks) {
        if (wrapper.getTaskType().isPresent() && !wrapper.getTaskType().get().isEmpty()) {
            return wrapper.getTaskType().get();
        } else {
            if (wrapper.getIssueInput().getField("parent") == null) {
                return defaultTaskTypeName;
            }
            return defaultIssueTypeForSubtasks;
        }
    }
}
